app/code/Magento/Catalog/Model/CategoryRepository.php

```diff
@@ -238,4 +241,26 @@ private function getMetadataPool()
         }
         return $this->metadataPool;
     }
+
+    /**
+     * Returns a cache key based on scope
+     *
+     * @param string|integer|null $storeId
+     *
+     * @throws SerializationException
+     * @return integer
+     */
+    private function getScopeCacheKey($storeId = null)
+    {
+        if (null !== $storeId && !is_numeric($storeId)) {
+            throw new SerializationException(
+                new Phrase(
+                    'The "%value" value\'s type is invalid. The "%type" type was expected. '
+                    . 'Verify and try again.',
+                    ['value' => $storeId, 'type' => 'int']
+                )
+            );
+        }
+        return $storeId ?? 'all';
```

Comment:
```suggestion
        return $storeId === null ? 'all' : (int)$storeId;
```
This will convert to int for sure